### ISSUE TITLE
Hide "id:Xxxx" when rendering take in list

### ISSUE THREAD

**Issue by flight16** [mpcjanssen/simpletask-android#281]
There is a plug-in that uses id:Nnn to link tasks to an outline. I want to use the prefix for linking to OmniFocus, another too application for Mac, but the IDs are Units that are long and ugly. Since this syntax is s todo.txt plug-in, it would be helpful if Simpleton was aware and ignored that token when displaying the task title in the list. I imagine it would still be displayed in edit mode. 

What do you think?  If you don't have the time to work on it, do you accept pull requests?

**Comment by mpcjanssen** [comment]
I do accept pull requests, so if you want to add this feel free. I am planning to make the rendering of the tasks more flexible by using the builtin Lua interpreter.
I can see the use of an invisible id: field though. Note that the task in the editor should always be an exact representation of the line in the todo.txt file.

**Comment by flight16** [comment]
Excellent.  What time frame do you anticipate the Lua interpreter enhancement?  Just want to make sure I don't write something that will soon become obsolete.

**Comment by mpcjanssen** [comment]
The Lua part will not happen any time soon.
But even if it does the task parser needs to be extended anyway. So all that work needs to be done. So I don't think any work would be lost. For now I would prefer an extra checkbox in the Other tab in the filter.

**Comment by flight16** [comment]
The checkbox you speak of would control whether id:xxx is included or not in the task list?  Wouldn't it be better to put it under settings, as it could apply to un-filtered lists, too?

**Comment by mpcjanssen** [comment]
Yes, it is similar to the other items on the Other tab in the filter.
I can imagine scenarios where you would want to save different filters with a different show id: setting. So best place is in the Other tab. 
An unfiltered list also has a filter (for sort order and other tab)

**Comment by flight16** [comment]
I'm new to Android development but want to work on this feature if I get time in the next month or two.  Are there any instructions for how to get this building?  What OS and toolchain do you use?

**Comment by mpcjanssen** [comment]
Importing into Android Studio should work fine for the cloudless version. Building the dropbox version requires a developer API key.

**Comment by mpcjanssen** [comment]
Related to #362

**Comment by mpcjanssen** [comment]
I want to implement this using a Lua callback which will give back an array of keys that will not be shown.

Default would be `id,meta`, where `meta` can be used by backends which are not file based to include an id to identify a task when it comes back as a string.
